### PR TITLE
Payment QRCode Detection and Parsing

### PR DESCRIPTION
**Pull request by a-szotyori**
**Please Note**: Tests will be added later this week. It would be great if we can release this as a beta version without tests to be able to use it in the showcase app.

BezahlCode and EPC069-12 (Stuzza (AT), GiroCode (DE)) is detected using Google's Mobile Vision API. Parsing is done according to the format specifications with IBAN validation and amount+currency normalization to fit the [Gini API format for amount](http://developer.gini.net/gini-api/html/entity_reference.html#extraction-entity-amount). 

### How to test
Run the example apps and scan invoices/remittance slips with the supported QRCode formats.

### Merging
Automatic

### Ticket 
Issues #126, #127

### CODE REVIEW

File: ginivision/src/main/java/net/gini/android/vision/camera/CameraActivity.java [hunk 1]
```diff
@@ -605,6 +606,11 @@ public void onDocumentAvailable(@NonNull Document document) {
         }
     }
 
+    @Override
+    public void onPaymentDataAvailable(@NonNull final PaymentData paymentData) {
+
```
Comment (kikettas) [inline ginivision/src/main/java/net/gini/android/vision/camera/CameraActivity.java]: This method should be implemented, right?

File: ginivision/src/main/java/net/gini/android/vision/internal/qrcode/AmountAndCurrencyNormalizer.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        }
+        try {
+            BigDecimal amountBigDecimal = new BigDecimal(amount.replace(",", "."));
+            if (amountBigDecimal.scale() < 2) {
+                amountBigDecimal = amountBigDecimal.setScale(2, BigDecimal.ROUND_HALF_EVEN);
+            }
+            return amountBigDecimal.toPlainString() + ":" + currency;
+        } catch (final NumberFormatException ignored) {
+        }
+        return "";
```
Comment (kikettas) [inline ginivision/src/main/java/net/gini/android/vision/internal/qrcode/AmountAndCurrencyNormalizer.java]: You can simplify this a little bit. 
```
if !TextUtils.isEmpty(amount) {
try ...
} else {
      return ""
}```

File: ginivision/src/main/java/net/gini/android/vision/internal/qrcode/QRCodeDetectorImpl.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        mHandlerThread = new HandlerThread("QRCodeDetectorThread",
+                Process.THREAD_PRIORITY_BACKGROUND);
+        mHandlerThread.start();
+        mHandler = new QRCodeDetectorHandler(mHandlerThread.getLooper(), qrCodeDetectorTask);
+    }
+
+    @Override
+    public void detect(@NonNull final byte[] image, @NonNull final Size imageSize,
+            final int rotation) {
+        if (mListener == null) {
```
Comment (kikettas) [inline ginivision/src/main/java/net/gini/android/vision/internal/qrcode/QRCodeDetectorImpl.java]: Here is `mHandler` not `mListener`, right?

File: ginivision/src/main/java/net/gini/android/vision/camera/CameraFragmentListener.java [hunk 4]
```diff
@@ -20,6 +21,8 @@
      */
     void onDocumentAvailable(@NonNull Document document);
```
Comment (kikettas) [inline ginivision/src/main/java/net/gini/android/vision/camera/CameraFragmentListener.java]: Don't forget to add the documentation before the final release!

File: ginivision/src/main/java/net/gini/android/vision/camera/CameraActivity.java (hunk 1 above)
Comment (a-szotyori) [reply ginivision/src/main/java/net/gini/android/vision/camera/CameraActivity.java]: Yes, I will add to the documentation that this has to be overriden in a subclass to get the payment data with the Screen API.

File: ginivision/src/main/java/net/gini/android/vision/internal/qrcode/QRCodeDetectorImpl.java (hunk 3 above)
Comment (a-szotyori) [reply ginivision/src/main/java/net/gini/android/vision/internal/qrcode/QRCodeDetectorImpl.java]: If there is no Listener I avoid processing any images. (If no one is interested why do the work? 😄). The handler cannot be null.

I will add a comment to clarify this.

**Pull request merged**